mlir/include/mlir/Dialect/Vector/Utils/VectorUtils.h

```diff
@@ -180,6 +180,23 @@ struct MaskableOpRewritePattern : OpRewritePattern<SourceOp> {
 /// are not linearizable.
 bool isLinearizableVector(VectorType type);
 
+/// Create a TransferReadOp from `source` with static shape `readShape`. If the
+/// vector type for the read is not the same as the type of `source`, then a
+/// mask is created on the read.
+/// enableMasking if false, the inBoundsVal values are set properly, based on
+///     the rank dimensions of the source and destination tensors.
```

Comment:
Please update this comment.

1. Fix formatting
2. "then a mask is created on the read." - the mask is never created if `enableMasking` is false. 
3. Missing note on `indices` being hard-coded as `0`.

Point 2. is rather counter-intuitive to me. @pashu123, you introduced `doMasking` in #88249. Looking at the implementation and you PR, it feels like you meant something like `bool useInBoundsInsteadOfMasking` rather than `doMasking`/`enableMasking`? As in:

BEFORE #88249:
```cpp
if (llvm::equal(readShape, sourceShape))
  // use masking
```

AFTER #88249:
```
if (llvm::equal(readShape, sourceShape) && !useInBoundsInsteadOfMasking)
  // use masking
```

Put differently, the extra logic added to `createReadOrMaskedRead ` in #88249 reads to me as:
*  "In some cases I want to disable masking and use in_bounds attr instead", rather than
* "I want a dedicated toggle to enable/disable masking".

I'm asking to better understand the current logic 😅

Reply:
"I feel the same way too. Using `useInBoundsInsteadOfMasking` provides a better understanding of the entire context." I can modify the variables.

Reply:
@banach-space To me this is just a potential optimization (maybe I'm wrong). It looks like the code is setting up for a later optimization to turn read_transfer into a simple load. But if we do masking or the bounds of a dimension is different, the optimization can't be done. It alsocan't be done if masking is done. So, it looked to me that if no masking and bounds the same, set it up for later pass to do load instead of a transfer (with skipping the masking op). Or just maybe just removing an unnecessary op - masking. Maybe I misunderstood the purpose of it too. That's what I thought... :)
The rest is fixed, I hope... :)
Thanks!

Reply:
> useInBoundsInsteadOfMasking

I renamed them, since I'm moving the code as utility methods. Thanks!